### PR TITLE
feat: add `useList` documentation

### PR DESCRIPTION
**Pull request by salihozdemir**
Added new `useList` documentation.

### CODE REVIEW

File: documentation/docs/api-reference/core/hooks/data/useList/index.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    },
+    //...
+};
+```
+
+### `successNotification`
+
+> [`NotificationProvider`](/api-reference/core/providers/notification-provider.md) is required.
+
+After data is fetched successfully, `useList` can call `open` function from [`NotificationProvider`][notification-provider] to show a success notification. With this prop, you can customize the success notification.
```
Comment (omeraplak) [inline documentation/docs/api-reference/core/hooks/data/useList/index.md:205]: these links are not working

File: documentation/docs/api-reference/core/hooks/data/useList/index.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+            data,
+        };
+    },
+    //...
+};
+```
+
+### `successNotification`
+
+> [`NotificationProvider`](/docs/api-reference/core/providers/notification-provider/) is required.
```
Comment (aliemir) [inline documentation/docs/api-reference/core/hooks/data/useList/index.md:203]:
```suggestion
> [`NotificationProvider`](/docs/api-reference/core/providers/notification-provider/) is required for this prop to work.
```

File: documentation/docs/api-reference/core/hooks/data/useList/index.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+            description: "Success with no errors",
+            type: "success",
+        };
+    },
+});
+```
+
+### `errorNotification`
+
+> [`NotificationProvider`](/docs/api-reference/core/providers/notification-provider/) is required.
```
Comment (aliemir) [inline documentation/docs/api-reference/core/hooks/data/useList/index.md:221]:
```suggestion
> [`NotificationProvider`](/docs/api-reference/core/providers/notification-provider/) is required for this prop to work.
```

**Review by omeraplak — APPROVED** [review]

**Review by alicanerdurmaz — APPROVED** [review]

File: documentation/docs/api-reference/core/hooks/data/useList/index.md [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+
+## Filtering
+
+`useList` hook supports filtering feature. You can pass `filters` property to enable filtering. In order to handle filtering, `useList` hook passes `filters` property to the `getList` method from the `dataProvider`.
+
+Dynamically changing the `filters` property will trigger a new request.
+
+<FilteringLivePreview />
+
+## `liveProvider`
```
Comment (aliemir) [inline documentation/docs/api-reference/core/hooks/data/useList/index.md:49]:
```suggestion
## Realtime
```

or `"Live Provider"` will look better I guess 🤔

Comment (aliemir) [reply documentation/docs/api-reference/core/hooks/data/useList/index.md:49]: @salihozdemir @omeraplak

**Review by omeraplak — APPROVED** [review]

**Pull request merged**